### ISSUE TITLE
Doc build in kellnr docker container fails with: Invalid cross-device link

### ISSUE THREAD

**Issue by mbodmer** [kellnr/kellnr#254]
What could be the cause for failing doc builds inside the container?

`Invalid cross-device link (os error 18)`, while trying to rename 
from `/root/.rustup/toolchains/...` to `/root/.rustup/tmp/...` 
does not make sense, because this is the same filesystem as I understand it.

Any ideas?

Thanks for kellnr, by the way. It's great!

```sh
Running `rustc --crate-name regex --edition=2021 
/root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/regex-1.10.4/src/lib.rs 
--error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat 
--crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no 
--cfg 'feature="std"' --cfg 'feature="unicode-perl"' 
-C metadata=acec26d0b77f6756 -C extra-filename=-acec26d0b77f6756 
--out-dir /opt/kdata/doc_queue/cRvqR6YjeT/mylib-sys-2.4.0/target/debug/deps 
-L dependency=/opt/kdata/doc_queue/cRvqR6YjeT/mylib-sys-2.4.0/target/debug/deps 
--extern regex_automata=/opt/kdata/doc_queue/cRvqR6YjeT/mylib-sys-2.4.0/target/debug/deps/libregex_automata-f6805926912f666c.rmeta 
--extern regex_syntax=/opt/kdata/doc_queue/cRvqR6YjeT/mylib-sys-2.4.0/target/debug/deps/libregex_syntax-5957eeb6d1504fb2.rmeta --cap-lints allow`
info: removing previous version of component 'cargo'
info: rolling back changes
error: could not rename component file from 
'/root/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/share/doc/cargo' to
'/root/.rustup/tmp/xrl8fy9awgv8lkcy_dir/bk': Invalid cross-device link (os error 18)
error: could not compile `proc-macro2` (build script)
```

**Comment by secana** [comment]
Hi. Thanks for reporting. 
Does this only happen with this specific crate? Is there anything special (other toolchain, edition)? 
I've never seen this error before, I need more information to replicate it.

**Comment by mbodmer** [comment]
Hi, thanks for the fast reply.

While researching the error I found this issue: https://github.com/rust-lang/rustup/issues/1239

I then tried to use "ubuntu:22.04" as the docker base image, which worked. Also I have successfully used the official "rust:1.77-bookworm" as base image, allowing to omit the rustup install step in your Dockerfile.

I have not further researched the problem, as this works for me. It seems to b a problem related to docker OverlayFs.
